src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,50 +1,96 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Prop = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie = ({ onAdd }: Prop) => {
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    description: '',
+  });
+
+  const emptyForms = newMovie.title.trim() === ''
+    || newMovie.imgUrl.trim() === ''
+    || newMovie.imdbUrl.trim() === ''
+    || newMovie.imdbId.trim() === '';
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+      description: '',
+    });
+  };
+
+  function sendInfo(event: React.FormEvent<HTMLFormElement>) {
+    event?.preventDefault();
+    setCount(count + 1);
+    onAdd({
+      title: newMovie.title,
+      description: newMovie.description,
+      imgUrl: newMovie.imgUrl,
+      imdbUrl: newMovie.imdbUrl,
+      imdbId: newMovie.imdbId,
+    });
+    reset();
+  }
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={sendInfo}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={newMovie.title}
+        onChange={(value) => setNewMovie({ ...newMovie, title: value })}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        onChange={(value) => setNewMovie({ ...newMovie, description: value })}
```

Comment:
it is redundant to make arrow function in jsx for every TextField
pass just reference on setNewMovie and change proper field with event.target.name
will be necessary to change prop onChange in TextField on such
`React.Dispatch<React.SetStateAction<strong>>` and pass name attribute to input
```suggestion
        onChange={setNewMovie}
```